app/views/aspects/_aspect_dropdown.html.haml

```diff
@@ -0,0 +1,44 @@
+-# locals: selected_aspects. Note: all_aspects is a global in the ApplicationController
+.btn-group.aspect_dropdown
+  %button.btn.btn-default.dropdown-toggle{!current_user.getting_started? ? {"data-toggle" => "dropdown"} : |
+  {"data-toggle" => "dropdown", |
+  :title => popover_with_close_html("2. #{t('shared.public_explain.control_your_audience')}"), |
+  "data-content" => t("shared.public_explain.visibility_dropdown")}}   |
+    - if public_selected?(selected_aspects)
+      %i.entypo-globe.small#visibility-icon
+      %span.text
+        = t("public")
+    - else
+      %i.entypo-lock.small#visibility-icon
+      %span.text
+        - if all_aspects_selected?(all_aspects, selected_aspects)
+          = t("all_aspects")
+        - elsif selected_aspects.size == 1
+          = selected_aspects.first.name
+        - else
+          = t("shared.aspect_dropdown.toggle", count: selected_aspects.size)
+    %span.caret
+  %ul.dropdown-menu.pull-right{unSelectable: "on"}
+
+    %li.public.radio{"data-aspect_id" => "public", class: ("selected" if public_selected?(selected_aspects))}
+      %a
+        %span.status_indicator
+          %i.glyphicon.glyphicon-ok
+        %span.text
+          = t("public")
+    %li.all_aspects.radio{"data-aspect_id" => "all_aspects", class: ("selected" if (!public_selected?(selected_aspects) |
+      && all_aspects_selected?(all_aspects, selected_aspects)))} |
+      %a
+        %span.status_indicator
+          %i.glyphicon.glyphicon-ok
+        %span.text
+          = t("all_aspects")
+    %li.divider
+    - for aspect in all_aspects
+      %li.aspect_selector{"data-aspect_id" => aspect.id, class: !all_aspects_selected?(all_aspects, selected_aspects) |
```

Comment:
- Don't mix styles in the same hash.
- Don't use the `|` character to split up lines. Wrap on commas or extract code into helper.